lib/temporal/interceptor/client.rb

```diff
@@ -0,0 +1,90 @@
+module Temporal
+  module Interceptor
+    class Client
+      StartWorkflowInput = Struct.new(
+        :workflow,
+        :args,
+        :id,
+        :task_queue,
+        :execution_timeout,
+        :run_timeout,
+        :task_timeout,
+        :id_reuse_policy,
+        :retry_policy,
+        :cron_schedule,
+        :memo,
+        :search_attributes,
+        :headers,
+        :start_signal,
+        :start_signal_args,
+        keyword_init: true,
+      )
+
+      DescribeWorkflowInput = Struct.new(
+        :id,
+        :run_id,
+        keyword_init: true,
+      )
+
+      QueryWorkflowInput = Struct.new(
+        :id,
+        :run_id,
+        :query,
+        :args,
+        :reject_condition,
+        :headers,
+        keyword_init: true,
+      )
+
+      SignalWorkflowInput = Struct.new(
+        :id,
+        :run_id,
+        :signal,
+        :args,
+        :headers,
+        keyword_init: true,
+      )
+
+      CancelWorkflowInput = Struct.new(
+        :id,
+        :run_id,
+        :first_execution_run_id,
+        :reason,
+        keyword_init: true,
+      )
+
+      TerminateWorkflowInput = Struct.new(
+        :id,
+        :run_id,
+        :first_execution_run_id,
+        :reason,
+        :args,
+        keyword_init: true,
+      )
+
+      def start_workflow(input)
```

Comment:
All of our other interceptors in all of our other SDKs allow you to choose whether to go to the next interceptor or not. These aren't a "linear" chain, they are a hierarchical one. I should be able to early return here and not go to the next interceptor if I want.

I think the Python model (which mimics Go and TS and others mostly) would be reasonable to follow here.

Reply:
You can do the same here by not calling `super`, it works exactly like you'd expect

Reply:
I see. So long as altering the input alters it for successive calls and I can react to their results, that's probably good enough.

Does seem easier from a programming POV not to have to loop for every interceptor invocation, but if the result is the same, no prob.

Reply:
Yep, you can modify the input and also the return value if you want to:

```ruby
class MyInterceptor  < Temporal::Interceptor::Client
  def start_workflow(input)
    input.headers = { 'AUTH_ID' => 'xxx' }
    result = super(input) # you don't have to pass input in, Ruby will use it if you just call super without args
    logger.log(result)
    result
  end
end
```